### DISCUSSION TITLE
I got Marlin 2.1.2 (now 2.1.3) bugfix running on the Neptune 3 with the MKS H43 :D

### DISCUSSION THREAD

**Discussion by TheFeralEngineer** [NARUTOfzr/Neptune_3#60] [Show and tell]
https://www.youtube.com/watch?v=bPO3dIXOnKk

**Comment by NARUTOfzr** [comment]
Haha, your theme song is really great. Glad to see you got the H43 screen working smoothly. I apologize for not being able to respond quickly to your questions. The Spring Festival is coming soon. There are many things to be busy, and some things will be delayed. Please don't mind.

**Reply by TheFeralEngineer** [reply]
No problem. I've worked for a few different Asian companies, so I understand all of the different festivals and such. 

I have the h43 working on both the pro firmware and the latest mainstream release, but I need to figure out creating a Z offset screen. I really don't like the DGUS workflow. I believe nextion has a better development environment. I prefer Klipper firmware, so I'm not intending on going much further with the H43, but it was definitely fun to get it working. I learned a lot. 

Is the raw nextion project from the pro with all of the assets available anywhere on GitHub? I have a nextion display I would like to try to load it on

**Comment by Eljefe227** [comment]
How did you manage to do it? I have the Neptune 3 with an orbiter extruder and an improved hot end but I am not able to compile Marlin 2.1 and have the input shaping work. I'm desperate 😢😭

**Reply by TheFeralEngineer** [reply]
to get stock marlin working was a very tedious endeavor and it was very alpha stage, to say the least. My recommendation to you would be to ditch the marlin idea and install klipper. I recommend the bigtreetech cb1 based products, as they seem to be incredibly reliable. I only had 2.1.x working long enough to provide proof of concept, I cannot confirm or deny that everything will work 100%.

**Reply by Eljefe227** [reply]
Unfortunately, it was what I feared, well, at least I'm left with the fact that it wasn't my problem. Thank you very much for the quick answer 😄👍

**Reply by TheFeralEngineer** [reply]
nah, you were in the right ballpark, but in order to make stock marlin work with the elegoo product, i had to do a lot of pin file editing. it's not hard, but it's not easy to explain without really digging in deep. marlin is a can of worms when it comes to doing stuff that isn't in configuration.h 😄

**Reply by Eljefe227** [reply]
Basically I took Marlin 2.1, modified the configs to make it work on Neptune 3 and pasted the resources from the source code that Naruto made. Right now I supposedly have Marlin 2.1 but when using the M593 command (I think it was) it does not recognize it. I've been literally seeing for a week if I'm able to do something about it but the truth is that not knowing the basic fundamentals of Marlin is all trial and error. Now the printer works fine but i want the input shaping .I'll have to switch to klipper

**Reply by TheFeralEngineer** [reply]
I highly recommend it. I run it on every printer I have and would never go back
